src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -1393,10 +1419,18 @@ private void Indent(bool beforeEndElement)
             else if (!_stack[_top].mixed)
             {
                 _textWriter.WriteLine();
-                int i = beforeEndElement ? _top - 1 : _top;
-                for (i *= _indentation; i > 0; i--)
+                int i = (beforeEndElement ? _top - 1 : _top) * _indentation;
+                if(i <= _indentChars.Length)
```

Comment:
Why do we want to special case `i <= _indentChars.Length`? The `while` loop below should be able to handle this case.